modules/lang-painless/src/main/java/org/elasticsearch/painless/node/SExpression.java

```diff
@@ -57,35 +48,6 @@ public AExpression getExpressionNode() {
 
     @Override
     public <Scope> void visitChildren(UserTreeVisitor<Scope> userTreeVisitor, Scope scope) {
-        expressionNode.visit(userTreeVisitor, scope);
-    }
-
-    @Override
-    void analyze(SemanticScope semanticScope) {
-        Class<?> rtnType = semanticScope.getReturnType();
-        boolean isVoid = rtnType == void.class;
-        boolean lastSource = semanticScope.getCondition(this, LastSource.class);
-
-        if (lastSource && !isVoid) {
-            semanticScope.setCondition(expressionNode, Read.class);
-        }
-
-        AExpression.analyze(expressionNode, semanticScope);
-        Class<?> expressionValueType = semanticScope.getDecoration(expressionNode, ValueType.class).getValueType();
-
-        boolean rtn = lastSource && isVoid == false && expressionValueType != void.class;
-        semanticScope.putDecoration(expressionNode, new TargetType(rtn ? rtnType : expressionValueType));
-
-        if (rtn) {
-            semanticScope.setCondition(expressionNode, Internal.class);
-        }
-
-        expressionNode.cast(semanticScope);
```

Comment:
Where did this cast move to in `DefaultSemanticAnalysisPhase`?

Reply:
Slight logic change:
It's now `decorateWithCast(userStatementNode, semanticScope);` and only applied if this value is returned; otherwise the cast is redundant under the old logic since we set both value type and target type to the same thing.